### PR TITLE
Add test for ApplicationProcessor classes

### PR DESCRIPTION
**Pull request by ChristopherAbram**
The unit test for `ApplicationProcessor` classes:
- I observed that `FilePickerProcessor` classes succeed when a path associated with an object is empty or point to wrong file
   - Success when path is empty -> should fail (return `false`),
   - Success when path points to non-existing file -> should fail (return `false`),
   - Some classes throws an exception when path points to file which exists but have wrong structure -> perhaps the exception should be caught in `process()` and the method should return false
- The `ApplicationProcessors::processCommands()` is not tested. The only idea I have is to verify exception handling. The thing is that this procedure relies on `SequenceProcessor` classes, so this test would mostly test dependencies, not the actual procedure.

📌Please give me hints about `FilePickerProcessor`, at the moment implementation requires some fixes.
❗ Because we don't use mocking approach, I created a test fixture which creates default objects (`default-constructible`) as stubs. However, this way breaks rules of unit testing with regards to isolating dependencies, which in turn makes them class-level integration tests.
❗ Many test fails at this stage.

### CODE REVIEW

File: src/tests/class_tests/smartpeak/source/ApplicationProcessor_test.cpp [hunk 1]
```diff
@@ -4,14 +4,112 @@
...
+#include <SmartPeak/core/Filenames.h>
 
 using namespace SmartPeak;
 using namespace std;
+namespace fs = boost::filesystem;
 
-BOOST_AUTO_TEST_SUITE(applicationprocessor)
+struct ApplicationProcessorFixture 
+{
+	/* ctor/dtor */
```
Comment (bertrandboudaud) [inline src/tests/class_tests/smartpeak/source/ApplicationProcessor_test.cpp]: You used tabs everywhere, it seems the other files are using 2 spaces indentation.

**Comment by bertrandboudaud** [comment]
For FilePickerProcessor I agree that for a non-existing file (or empty), it should return false.
However; FilePickerProcessor may be used to write a file (?) in that case maybe not...

About the Exception ... this is not easy, don't know if it has already been discussed.. for me, an exception should be caught only at some point if the catcher can handle it: he knows what is the exception, how critical it is. a general rule (catch everything in process()). I am not sure. It's probably nice to catch most of the time, but may be dedicated catch would be nice. So the rule would be : if the Processor can catch the error and treat it without putting the data model into a bad state, yes, catch and return false (again, it should be most of the case). else throw.

File: src/tests/class_tests/smartpeak/source/ApplicationProcessor_test.cpp [hunk 2]
```diff
@@ -4,14 +4,112 @@
...
+public:
+	ApplicationHandler ah_;
+	std::string datapath_;
+	Filenames filenames_;
+};
+
+BOOST_FIXTURE_TEST_SUITE(ApplicationProcessor, ApplicationProcessorFixture)
+
+/* processCommands */
+//BOOST_AUTO_TEST_CASE(buildcommandsfromids)
```
Comment (dmccloskey) [inline src/tests/class_tests/smartpeak/source/ApplicationProcessor_test.cpp]: Is there a reason for the commented code here?

**Review by dmccloskey — CHANGES_REQUESTED** [review]
Great work getting this tests in place!  The test fixture is a fine way to reduce redundant loading of the class.  The use of other dependencies to load in the data, etc. is fine from my point of view given the complexity (or lack thereof) of this software application so long as there is test coverage across all dependencies in such a way that the fault can be quickly isolated if something goes wrong.

I agree with Bertrand in regards to the questions about the `FilePickerProcessor` class.

Please be sure to change the spacing to two spaces instead of tabs.

Please add in the unit test for ApplicationProcessors::processCommands() but you can omit a test for a value.  We can simply check to make sure it runs without throwing an error.

File: src/tests/class_tests/smartpeak/source/ApplicationProcessor_test.cpp (hunk 1 above)
Comment (ChristopherAbram) [reply src/tests/class_tests/smartpeak/source/ApplicationProcessor_test.cpp]: I guess I should have changed default settings in Visual Studio 2019 ¯\\_(ツ)_/¯

File: src/tests/class_tests/smartpeak/source/ApplicationProcessor_test.cpp (hunk 2 above)
Comment (ChristopherAbram) [reply src/tests/class_tests/smartpeak/source/ApplicationProcessor_test.cpp]: No need for it, this is still WIP

**Review by ghost — COMMENTED** [review]
Looking good from my side once CI for Ubuntu are passed.

**Review by dmccloskey — APPROVED** [review]

**Pull request merged**